src/crl.rs

```diff
@@ -46,6 +46,60 @@ pub trait CertRevocationList: Sealed {
         supported_sig_algs: &[&SignatureAlgorithm],
         issuer_spki: &[u8],
     ) -> Result<(), Error>;
+
+    /// Returns true if the CRL can be considered authoritative for the given certificate.
+    ///
+    /// A CRL is considered authoritative for a certificate when:
+    ///   * The certificate issuer matches the CRL issuer and,
+    ///     * The certificate has no CRL distribution points, and the CRL has no issuing distribution
+    ///       point extension.
+    ///     * Or, the certificate has no CRL distribution points, but the the CRL has an issuing
+    ///       distribution point extension with a scope that includes the certificate.
+    ///     * Or, the certificate has CRL distribution points, and the CRL has an issuing
+    ///       distribution point extension with a scope that includes the certificate, and at least
+    ///       one distribution point full name is a URI type general name that can also be found in
+    ///       the CRL issuing distribution point full name general name sequence.
+    ///
+    /// In all other circumstances the CRL is not considered authoritative.
+    fn authoritative_for(&self, cert: &Cert<'_>) -> bool {
+        // In all cases we require that the authoritative CRL have the same issuer
+        // as the certificate. Recall we do not support indirect CRLs.
+        if self.issuer() != cert.issuer() {
+            return false;
+        }
+
+        match (
+            cert.crl_distribution_points(),
+            self.issuing_distribution_point(),
+        ) {
+            // If the certificate has no CRL distribution points, and the CRL has no issuing distribution point,
+            // then we can consider this CRL authoritative based on the issuer matching.
+            (None, None) => true,
```

Comment:
We can group these two arms: `(cert_dps, None) => cert_dps.is_none()`. That seems like a more compact representation of the original intent.

Reply:
Done.